pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt

```diff
@@ -36,6 +36,15 @@ class PushNotificationsAPI(private val instanceId: String) {
   var deviceId: String? = null
   var fcmToken: String? = null
 
+  // Handles the JsonSyntaxException properly
+  private fun safeExtractJsonError(possiblyJson: String): NOKResponse {
+    return try {
+      gson.fromJson(possiblyJson, NOKResponse::class.java)
+    } catch (jsonException: JsonSyntaxException) {
+      unknownNOKResponse
```

Comment:
Would it be better if we would insert description of `JsonSyntaxException` into error message?

Reply:
I don't know. If we are not reporting back the exceptions, not sure if there's anything useful to do in here...

Reply:
I'll add logging here.